pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,28 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+// LeaderElectionConfiguration defines the configuration of leader election
+// clients for components that can run with leader election enabled.
+type LeaderElectionConfiguration struct {
+	// leaderElect enables a leader election client to gain leadership
+	// before executing the main loop. Enable this when running replicated
+	// components for high availability.
+	LeaderElect bool `json:"leaderElect"`
+	// leaseDuration is the duration that non-leader candidates will wait
+	// after observing a leadership renewal until attempting to acquire
```

Comment:
This first sentence is a little hard to comprehend. Can't it be something as simple as "leaseDuration is the maximum duration between leadership renewals. If the incumbent leader fails to renew its leadership before this deadline, non-leaders attempt to acquire the leadership". Or something like that.